docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,190 @@
+[seven day limit]: https://github.com/alphagov/email-alert-api/blob/f0c61539f7244bdb52db54d277a46412b728d642/app/services/auth_token_generator_service.rb#L7
+
+## Status
+
+Accepted
+
+## Consequences
+
+### Historic deletion worker
+
+We implemented a [worker] which runs periodically every day at [midday].
+Removing this data was deemed as non urgent and so deleting once a day, in
+hours, seemed sufficient.
+
+### The first historic deletion run
+
+On the first run of the worker which handles deleting historical data we
+removed over 33 million records. A breakdown of the amount of deletions per
+model can be found below:
+
+* ContentChange - 148,617
+
+* MatchedContentChange - 1,825,983
+
+* Message - 15
+
+* MatchedMessage - 16,099
+
+* DigestRun - 677
+
+* DigestRunSubscriber - 30,858,770
+
+* Subscription - 1,046,576
+
+* SubscriberList - 11,470
+
+* Subscriber - 196,376
+
+### Restrict analytics to 1 year of unused data
+
+We now restrict our unused data for analytics to a period of at most 1 year.
+This means any queries looking for this data (e.g. ended subscription's over a
+year old) will be met with no results. We don't expect this to be an issue due
+to our current expectation of the data we'll need for future analytics.
+
+### Modify the nullify process and retire the deactivated concept
+
+When documenting our data retention policy as described in this ADR, we found
+it difficult to understand and document the tangential concept of nullifying /
+deactivating subscribers.
+
+To give some context on these concepts:
+
+* Nullifying subscribers means to update the `address` of `Subscriber` records
+  which have been unsubscribed for more than 28 days to `nil`.
+
+* Deactivating subscribers is a concept that revolves around the
+  `deactivated_at` field on the subscriber model. This field is populated when
+  a subscriber unsubscribes from their last active subscription. It is used as
+  a means to determine when a subscriber can be nullified.
+
+We can however obtain the information held within `deactivated_at` through the
+`ended_at` field of the most recent ended subscription of the subscriber,
+meaning the deactivated concept is superfluous.
+
+Subsequently we didn't want to spend energy documenting a process that didn't
+make much sense to begin with so it was decided that it would be easier (and
+beneficial as a whole) to just [remove] the deactivated concept completely.
+
+In removing the deactivated concept we [modified the nullify process] to use
+the `ended_at` field of the last subscription, instead of `deactivated_at`. We
+also updated the process to nullify subscribers which have no subscriptions
+(active or ended) and are older than 28 days. This prevents the system from
+storing user's personal data longer than needed, if for whatever reason, a
+subscriber ends up with no subscriptions. See examples of where this happened
```

Comment:
```suggestion
subscriber is created without any subscriptions. See examples of where this happened
```

We should be more precise to distinguish between a subscriber with "no (active) subscriptions" and a subscriber that has never had any subscriptions, which is what we mean here.